spinach/brokers/redis.py

```diff
@@ -272,19 +289,34 @@ def enqueue_jobs_from_dead_broker(self, dead_broker_id: uuid.UUID) -> int:
             self._to_namespaced(ALL_BROKERS_HASH_KEY),
             self._to_namespaced(ALL_BROKERS_ZSET_KEY),
             self.namespace,
-            self._to_namespaced(NOTIFICATIONS_KEY)
+            self._to_namespaced(NOTIFICATIONS_KEY),
+            self._to_namespaced(MAX_CONCURRENCY_KEY),
+            self._to_namespaced(CURRENT_CONCURRENCY_KEY),
         )
 
     def register_periodic_tasks(self, tasks: Iterable[Task]):
         """Register tasks that need to be scheduled periodically."""
-        tasks = [task.serialize() for task in tasks]
+        _tasks = [task.serialize() for task in tasks]
         self._number_periodic_tasks = len(tasks)
         self._run_script(
             self._register_periodic_tasks,
             math.ceil(datetime.now(timezone.utc).timestamp()),
             self._to_namespaced(PERIODIC_TASKS_HASH_KEY),
             self._to_namespaced(PERIODIC_TASKS_QUEUE_KEY),
-            *tasks
+            *_tasks
+        )
+        self.set_concurrency_keys(tasks)
```

Comment:
The `tasks` argument that `register_periodic_tasks` receives contains only periodic tasks.

Let's separate `register_periodic_tasks` and `set_concurrency_keys`, adding the later as an abstract method to the base `Broker` and calling it explicitly from the `Engine._arbiter_func`.

Reply:
I was thinking the same actually, it felt like a hack after you suggested it.